lib/passport-configurator.js

```diff
@@ -65,6 +65,24 @@ PassportConfigurator.prototype.setupModels = function(options) {
   if (!this.userCredentialModel.relations.user) {
     this.userCredentialModel.belongsTo(this.userModel, {as: 'user'});
   }
+  _.each(_.keys(options), (option) => {
+    switch (option) {
+      case 'userModel':
+      case 'userCredentialModel':
+      case 'userIdentityModel':
+        break;
+      default:
+        let sOption = option.split('Model');
```

Comment:
I would prefer to have `options` to be more explicit:
```js
{
  userModel: ...,
  userCredentialModel: ...,
  userIdentityModel: ...,
  additionalModels: {
    anotherUserRelatedModel: ...
  }
}
```

Reply:
Is this preference something that needs to be implemented for this PR or was it to open discussion about implementation?
When initially looking to implement this functionality I found multiple posts online with people expecting it to work the way implemented in this PR. I also expected it to work and was surprised when it was not. Considering there are only three models required and anything else would be inherently additional models; implementing additional models under an additionalModels property seemed like it would require additional error checking to ensure names do conflict with userModel, userCredentialModel, and userIdentityModel whereas making use of the default switch statement takes care of it.
There are also no additional options associated with the setupModels method other than models to be set up, so to me the implementation in this PR makes sense. Explicitly putting them under an additional property seems kind of redundant to me.

Reply:
@raymondfeng Makes sense. Perhaps it would be a good idea to have both additionalModels and additionalUserModels. In this PR any additional models are given a belongsTo userModel relation if does not already exist. There could be use cases where people want to attach models that are not related to users though. For example having an Account model which hasMany users.

What do you think?

Reply:
@raymondfeng ?

Reply:
@raymondfeng Here is an example as requested.
```
{
  userModel: ...,
  userCredntialModel: ...,
  userIdentityModel: ...,
  additionalUserModels: {
    userRelatedModel: ...
  },
  additionalModels: {
    nonUserRelatedModel: ...
  }
}
```
For the additionaUserModels the current logic would remain which ensures a belongsTo User relation is in place. For additionalModels no relations would be modified.

Reply:
What code will leverage `additionalModels`?

Reply:
@raymondfeng The verifyMethod callback. An example use case would be to have an Account model which the user belongs to. I am currently using this use case in order to retrieve the user profile and account details from a custom passport strategy which then the verifyMethod ensures the account record exists and that the user belongs to that account.

Reply:
So the `additionalModels` just serves as a registry for `verifyMethod` to access models such as `Account`? If so, why not using `userModel.app.models.Account` instead?

Reply:
Initially when implementing I came across multiple posts with people expecting this type of implementation to work so I kept with that that route. Outside of that I think it would be easier for instructions to people implementing custom passport modifications. For instance, if someone builds a library which incorporates the verifyMethod callback, then it could remain untouched while allowing users implementing the library to just configure the appropriate models. I think this would make it a bit more flexible and easier to configure.

Reply:
@raymondfeng any thoughts?

Reply:
ok, so you want to have one more level of redirection? If so, I'm fine with that. Maybe you can have a default as `userModel.app.models.<your-model-name>`?

Reply:
@raymondfeng Yes, both options would be available. Making the model available as essentially a shortcut within this module, as well as if users choose to use userModel.app.models.<mode-name> would both work. The reason for the shortcut is because I saw various posts with people expecting it to work with adding additional models to the config. I expected it to operate that way as well and was surprised that it did not function. It will also allow for making additional implementations easier for people. If someone creates an additional loopback passport strategy that requires access to models, instead of hard coding it any required models can be configured. If users extend a model with their own implementation they will be able to use their own model by just configuring it appropriately.
If your good with this, I will make the necessary changes over the weekend so that you can complete the review first thing next week.

Reply:
Ok, that's good then. Thank you for your patience.

Reply:
@raymondfeng Discussed changes have been submitted for your review